diff --git a/cRequestHandler.cpp b/cRequestHandler.cpp
index 9f0552f..31400ef 100644
--- a/cRequestHandler.cpp
+++ b/cRequestHandler.cpp
@@ -546,7 +546,7 @@ string cRequestHandler::recordingsToXml(bool deleted) {
             string filesize = intToString(rec->FileSizeMB());
             string deleted = timeToString(rec->Deleted());
             string ichannelid = string(info->ChannelID().ToString());
-            string ichannelname = string(info->ChannelName());
+            string ichannelname = string(info->ChannelName() ? info->ChannelName() : "");
             string ititle = string(info->Title() ? info->Title() : "");
             string ishorttext = string(info->ShortText() ? info->ShortText() : "");
             string idescription = string(info->Description() ? info->Description() : "");
@@ -1052,4 +1052,4 @@ int cRequestHandler::handle404Error() {
     ret = MHD_queue_response(this->connection, MHD_HTTP_NOT_FOUND, response);
     MHD_destroy_response (response);
     return ret;
-}
\ No newline at end of file
+}
